.travis.yml

```diff
@@ -196,6 +196,8 @@ before_install:
           ldapadd -h localhost:3389 -D cn=admin,dc=symfony,dc=com -w symfony -f src/Symfony/Component/Ldap/Tests/Fixtures/data/fixtures.ldif
       fi
 
+    - if [[ $PHP = nightly ]]; then composer self-update --snapshot --1; fi
```

Comment:
passing both `--snapshot` and `--1` is useless. There is a single snapshot channel (and it is now about composer 2). `--1` means using the stable channel but restricted to `1.x` versions.

Reply:
if you pass multiple options, the first one will win in the order specified in https://github.com/composer/composer/blob/c6d0309d48e57c8e9a46988bd71d34c4079256a6/src/Composer/SelfUpdate/Versions.php#L24

Reply:
see https://github.com/composer/composer/issues/9189 for a discussion on Composer about having a snapshot of 1.x